### PR TITLE
#83: added method that get the adapter version into AdapterFactory

### LINKED ISSUE #83: Add a method that allows querying the version, to the `AdapterFactory`
The `AdapterFactory` should have a method that allows querying the version, so that `virtual-schema-common` can query the version before instantiating an adapter.

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #83

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.exasol</groupId>
     <artifactId>virtual-schema-common-java</artifactId>
-    <version>5.2.1</version>
+    <version>5.3.1</version>
```
Comment (AnastasiiaSergienko) [inline pom.xml]: I'm not sure if I should change the major version? It makes adapters implement the new method

Comment (redcatbear) [reply pom.xml]: Changing the minor version is correct. While it adds a new function, I leaves the existing interface intact. No breaking change --> no change in major number.

File: src/main/java/com/exasol/adapter/AdapterFactory.java [hunk 2]
```diff
@@ -19,4 +19,11 @@
      * @return new instance
      */
     public VirtualSchemaAdapter createAdapter();
+
+    /**
+     * Get a version of the adapter
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/AdapterFactory.java]:
```suggestion
     * Get the version of the Virtual Schema Adapter
```

Comment (redcatbear) [reply src/main/java/com/exasol/adapter/AdapterFactory.java]: I just realized, that we should also display the newly available adapter version number in a log message. Preferably in the same one where we already show the version number of the common part.

File: src/main/java/com/exasol/adapter/AdapterFactory.java [hunk 3]
```diff
@@ -19,4 +19,18 @@
      * @return new instance
      */
     public VirtualSchemaAdapter createAdapter();
+
+    /**
+     * Get the version of the Virtual Schema Adapter
+     *
+     * @return adapter version
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/AdapterFactory.java]:
```suggestion
     * @return Virtual Schema Adapter version
```

File: src/main/java/com/exasol/adapter/AdapterRegistry.java [hunk 4]
```diff
@@ -65,7 +65,10 @@ public void registerAdapterFactory(final String adapterName, final AdapterFactor
      */
     public VirtualSchemaAdapter getAdapterForName(final String name) {
         if (hasAdapterWithName(name)) {
-            return this.registeredFactories.get(name).createAdapter();
+            AdapterFactory factory = this.registeredFactories.get(name);
+            LOGGER.config("Virtual Schema Adapter name: " + factory.getAdapterName());
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/AdapterRegistry.java]: This is not the right place and things like this should also always be in a single message.
Add it to
`com.exasol.adapter.RequestDispatcher.logVersionInformation()` please.
You have to move the call a few lines down to the point where the adapter is loaded then.

File: src/main/java/com/exasol/adapter/AdapterFactory.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/AdapterFactory.java]: fixed

**Pull request merged**